### ISSUE TITLE
Not a bug report :/ Sorry, but where do I go for LAMP?

### ISSUE THREAD

**Issue by alanfluff** [WhitewaterFoundry/Pengwin#424]
Sorry, this is not a bug :/ But I didn't know where to go for docs and took a stab at posting here. Please move or ask me to repost elsewhere if needs be.

<hr />

I read:

> [Pengwin is particularly useful for frontend developers new to Linux](https://www.pengwin.dev/)

after trying free Ubuntu twice with two different docs on how best to get a web dev LAMP setup and each time getting unsatisfactory results.

I love the idea of paying for Linux and love that quote up there, but where do I go to find the docs on how best to get the components of LAMP installed, and where the config files (for Apache for example) are edited?

I hope I am not expecting too much of this purchase; maybe there are no docs of that sort?

I know I can research installing LAMP on Linux and get tons of hits, but I was looking for a safe, well documented path that matched my ver of Linux.

I am not as much of a newbie as I have suggested here, but I am also not a day to day user of Linux. I login to AWS to edit an httpd.conf once in a while, I used Mac and [CLAMP](https://github.com/jide/clamp) a lot before I moved to Win10.

Thanks a lot in advance for any help/suggestions. Cheers.

**Comment by sirredbeard** [comment]
Thiis kind of issue is fine. We hear you loud and clear. We are aiming for a one-click LAMP stack in our next major round of new features and bugs, 1.3. First we have to tackle a bug with MySQL in WSL but we're making headway. #394 

Someone from the community might be able to chime in here and give a brief rundown in the interim. The trick is you can't use systemd, so you're going to have to start and stop everything manually. It's not that hard but most tutorials these days assume systemd. Once you get it all set up and configured you can then automatically start them with our implementation of rc.local, which came out in 1.2.

**Comment by sirredbeard** [comment]
#387

**Comment by crramirez** [comment]
Hello,

Indeed any tutorial for Debian should work. Look for Debian buster. Probably the easiest way is to install Apache, PHP, and MySQL using apt install.

What is cool in pengwin is that with pengwin-setup you can install rc.local support in the services menu. And you can write the service xxx start in the /etc/rc.local and your LAMP will start when you start pengwin ok start menu.

Also, you can install Visual Studio Code which is used a lot by front end developers. And see your changes in your Apache and everything inside WSL. You will need an X server as well for vs code.

Sure you will receive more answers from the community. 

Just try it. If you fail just reset pengwin and start again.

Regards

**Comment by crramirez** [comment]
This tutorial looks good:

https://computingforgeeks.com/install-lamp-stack-on-debian-10-buster/

Skip step 2 until we fix the issue about MySQL.

Each time they say sudo systemctl replace it by services command. For example to start apache:

sudo service apache start 

Ask here we can help.

**Comment by alanfluff** [comment]
Wow.

I am *so* glad I paid for this Linux.

Thanks all, I am grateful and impressed and off to go try your suggestions.

PS: I've left this 'bug' open, so more may be posted, but if it's a pain being open (hurting some stats or other) then feel free to close or ask me to close.

**Comment by crramirez** [comment]
BTW to install MySQL use: 

`sudo apt update && sudo apt install -t unstable mysql-server`

But only after this bug be fixed: https://github.com/WhitewaterFoundry/Pengwin/issues/394

**Comment by sirredbeard** [comment]
No, please leave it open for now as a resource for others, until we implement the LAMP stack feature.

**Comment by goranovs** [comment]
Here is one good tutorial about installing PHP 7.3 https://tecadmin.net/install-php-on-debian-10-buster/
It works like a charm. The above tutorial in https://github.com/WhitewaterFoundry/Pengwin/issues/424#issuecomment-484341885 does not work for me. 

Also, I was able to install MySQL 5.7 with the instructions provided in https://github.com/WhitewaterFoundry/Pengwin/issues/424#issuecomment-484670054
`sudo apt update && sudo apt install -t unstable mysql-server`

Starting the MySQL with the instructions from https://github.com/WhitewaterFoundry/Pengwin/issues/394#issuecomment-481763707 works for me with a simple alias. 
```bash
alias mysql.start="sudo mv /etc/profile /etc/profile_ && sudo service mysql start; sudo mv /etc/profile_ /etc/profile"
```
In addition, I prefer to remove the MySQL password as it is very annoying when you need to configure multiple projects:
```
# Remove password
mysql
ALTER USER 'root'@'localhost' IDENTIFIED WITH mysql_native_password BY '';
FLUSH PRIVILEGES;

# Check authentication type native or socket
mysql
SELECT user,authentication_string,plugin,host FROM mysql.user;
```

----
_P.S. Maybe full "one-click LAMP" is not needed. Will be enough and cool if you guys add PHP with Composer & MySQL/MariaDB in the `pengiwn-setup` under the  "Programming" or "Services" tab (After you fix the MySQL bug). In most cases, Apache or Nginx are not needed for development. People using WSL/Pengwin as a development environment are using language-specific built-in servers. Laravel and Symfony, for example, are using PHP's server. Also Ruby on Rails (rails s), NodeJS & etc. Apache also can be an option under the  `pengiwn-setup`  if someone really needs to serve static content(HTML, CSS & etc.) or pure PHP script under WSL, but this looks like overkill for development in my opinion_

**Comment by sirredbeard** [comment]
Closing as LAMP Stack was added in 1.2.6.

**Comment by alanfluff** [comment]
It would be great to get a link to the docs that explain how to get LAMP setup, now it's included.

I looked after I saw this thread but could not find steps. Will go look again, and will answer back here if I find them.

Thanks in advance anyone, if you know and add the answer here.

**Comment by crramirez** [comment]
Hello,

Go to pengwin-setup -> SERVICES -> LAMP

Regards

**Comment by alanfluff** [comment]
My bad, I asked that question poorly ... I **have** LAMP installed via pengwin-setup.

What I don't have is some starting commands to get into configuring Apache (where is/are the setup files), is the M in lamp MySQL or MariaDB? What commands to begin setup?

I have configured command-line LAMPs before, but each (Mac, AWS, etc) has had a different set of commands.

I guess I am looking for a high-level walkthrough, so I know enough to open a text editor and begin configuring Apache, know the right initial command for getting in to the database component, etc.

Once I know enough to start I can Google the rest.

Thanks in advance for help and sorry that I don't know this interim step and need to bother asking here.

**Comment by alanfluff** [comment]
Re [above](https://github.com/WhitewaterFoundry/Pengwin/issues/424#issuecomment-557889437)...

For example, I dove in just now to see if I can find out some portions by trial and error.

MySQL docs plus poking about eventually led me to look in
`/var/log/mysql/error.log`
in an attempt to discover what temp password root MySQL had been assigned during setup, when I `more`ed the file I saw this:

```
2019-09-28 20:38:03 0 [ERROR] InnoDB: Linux Native AIO interface is not supported on this platform. Please check your OS documentation
and install appropriate binary of InnoDB.
2019-09-28 20:38:03 0 [Warning] InnoDB: Linux Native AIO disabled.
```

Perhaps Pengwin setup MariaDB and I am looking in the wrong place?

Thanks in advance for help with getting LAMP up.

**Comment by alanfluff** [comment]
( just remembered, MariaDB is a drop-in replacement and so perhaps commands are identical to MySQL.., not the best part of my above comment, but I am still stuck re [the prev' comment](https://github.com/WhitewaterFoundry/Pengwin/issues/424#issuecomment-557889437).
